device/core/devdoc/device_twin_requirements.md

```diff
@@ -3,146 +3,75 @@
-
-**SRS_NODE_DEVICE_TWIN_18_043: [** If the GET operation fails, `fromDeviceClient` shall call the done method with the error object in the first parameter **]**
-
-**SRS_NODE_DEVICE_TWIN_18_034: [** `fromDeviceClient` shall ignore any PATCH operations that arrive before the GET returns **]**
-
-**SRS_NODE_DEVICE_TWIN_18_012: [** `fromDeviceClient` shall remove the handlers for both the `subscribed` and `error` events before calling the `done` callback. **]**
-
-
-### _sendTwinRequest method (private)
-`_sendTwinRequest` is a helper method which sends an arbitrary message to the hub and waits for a response (or timeout or error)
+### constructor(transport: Client.Transport);
 
-**SRS_NODE_DEVICE_TWIN_18_014: [** `_sendTwinRequest` shall use an arbitrary starting `rid` and increment it by one for every call to `_sendTwinRequest` **]** 
+**SRS_NODE_DEVICE_TWIN_16_001: [** The `Twin` constructor shall subscribe to the `onDesiredPropertiesUpdate` event off the `transport` object. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_015: [** `_sendTwinRequest` shall add a handler for the `response` event on the twinReceiver object.  **]**
+### get(callback: (err: Error, twin?: Twin) => void): void;
 
-**SRS_NODE_DEVICE_TWIN_18_016: [** `_sendTwinRequest` shall use the `sendTwinRequest` method on the transport to send the request **]** 
+**SRS_NODE_DEVICE_TWIN_16_002: [** The `get` method shall call the `getTwin` method of the `Transport` object with a callback. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_017: [** `_sendTwinRequest` shall put the `rid` value into the `properties` object that gets passed to `sendTwinRequest` on the transport **]** 
+**SRS_NODE_DEVICE_TWIN_16_003: [** If the callback passed to the `getTwin` method is called with an error, the `callback` passed to the call to the `get` method shall be called with that error. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_018: [** The response handler shall ignore any messages that don't have an `rid` that matches the `rid` of the request **]**  
+**SRS_NODE_DEVICE_TWIN_16_004: [** If the callback passed to the `getTwin` method is called with no error and a `TwinProperties` object, these properties shall be merged with the current instance properties. **]**
 
-**SRS_NODE_DEVICE_TWIN_18_019: [** `_sendTwinRequest` shall call `done` with `err`=`null` if the response event returns `status`===200 or 204 **]**  
+**SRS_NODE_DEVICE_TWIN_16_005: [** Once the properties have been merged the `callback` method passed to the call to `get` shall be called with a first argument that is `null` and a second argument that is the current `Twin` instance (`this`). **]**
 
-**SRS_NODE_DEVICE_TWIN_18_020: [** `_sendTwinRequest` shall call `done` with an `err` value translated using `translateError`  **]** 
+**SRS_NODE_DEVICE_TWIN_16_006: [** For each desired property that is part of the `TwinProperties` object received, an event named after the path to this property shall be fired. **]**
```

Comment:
>.  [](start = 176, length = 2)

Does this event have  a parameter?  If it does, does it come from the PATCH or from the current instance?

Reply:
yes it's the actual property value. 

---
In reply to: [171956077](https://github.com/pierreca/azure-iot-sdk-node/pull/2#discussion_r171956077) [](ancestors = 171956077)